state/firewallrules.go

```diff
@@ -86,59 +110,66 @@ func NewFirewallRules(st *State) *firewallRulesState {
 
 // Save stores the specified firewall rule.
 func (fw *firewallRulesState) Save(rule FirewallRule) error {
-	if err := rule.WellKnownService.validate(); err != nil {
+	if err := checkModelActive(fw.st); err != nil {
 		return errors.Trace(err)
 	}
-	for _, cidr := range rule.WhitelistCIDRs {
+	buildTxn := func(int) ([]txn.Op, error) {
+		return fw.GetTransactions(rule, false)
+	}
+	if err := fw.st.db().Run(buildTxn); err != nil {
+		return errors.Annotate(err, "failed to create firewall rules")
+	}
+
+	return nil
+}
+
+func (fw *firewallRulesState) GetTransactions(rule FirewallRule, migrationWanted bool) ([]txn.Op, error) {
+	if err := rule.WellKnownServiceType().validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, cidr := range rule.WhitelistCIDRs() {
 		if _, _, err := net.ParseCIDR(cidr); err != nil {
-			return errors.NotValidf("CIDR %q", cidr)
+			return nil, errors.NotValidf("CIDR %q", cidr)
 		}
 	}
-	serviceStr := string(rule.WellKnownService)
+	serviceStr := string(rule.WellKnownServiceType())
```

Comment:
This is string to `WellKnownServiceType` and back to string.

There should be one member, `WellKnownService` of type `WellKnownServiceType` and one getter returning the same type. All other conversion should be executed as needed for params on the wire or the Mongo doc.